### DISCUSSION TITLE
Idea for major update Ringmode spacer

### DISCUSSION THREAD

**Discussion by KlashKontakt** [ryukau/VSTPlugins#47] [Ideas]
Hello Ryukau!

I hope you ok! Love your instruments! And thank you for your creativity!

So! I have some idea about for Ringmode spacer.
The idea is to incorporate stereo widening based on the Haas effect, where within the plugin, a time delay on one channel interacts with the opposite, much like a ring modulator side-chain effect, to avoid phase distortions in the center. That's the first point.

Secondly, we've previously discussed certain artifacts that might affect lower frequencies. To address such artifacts, I came across the Lanczos algorithm. Could this be useful in this tool?

Once again, thank you for your amazing tools.

Have nice day!

**Comment by ryukau** [comment]
Hi, @KlashKontakt.

Could you precisely define what phase distortion means here? Using Haas effect to avoid phase distortion sounds contradicting.

Would you elaborate on how to apply Lanczos algorithm? I read [Wikipedia page](https://en.wikipedia.org/wiki/Lanczos_algorithm), but it seem like an algorithm related to eigenvalues. I have no idea how it's useful for RingmodSpacer.

To be clear, I won't likely continue to work on RingmodSpacer unless your explanation on Lanczos algorithm has something neat. As [I said before](https://github.com/ryukau/VSTPlugins/issues/32#issuecomment-1476911843), there's already a solution to remove distortions and they are called limiter.

**Comment by ryukau** [comment]
Closed due to no response.
